tests/test_sparse_embeddings.py

```diff
@@ -1,124 +1,124 @@
+#             2.803262710571289,
+#             2.1904349327087402,
+#             1.0531445741653442,
+#         ],
+#     }
+# }
+
+# docs = ["Hello World"]
+
+
+# def test_batch_embedding():
+#     is_ci = os.getenv("CI")
+#     docs_to_embed = docs * 10
+
+#     for model_name, expected_result in CANONICAL_COLUMN_VALUES.items():
+#         model = SparseTextEmbedding(model_name=model_name)
+#         result = next(iter(model.embed(docs_to_embed, batch_size=6)))
+#         assert result.indices.tolist() == expected_result["indices"]
+
+#         for i, value in enumerate(result.values):
+#             assert pytest.approx(value, abs=0.001) == expected_result["values"][i]
+#         if is_ci:
+#             delete_model_cache(model.model._model_dir)
+
+
+# def test_single_embedding():
+#     is_ci = os.getenv("CI")
+#     for model_name, expected_result in CANONICAL_COLUMN_VALUES.items():
+#         model = SparseTextEmbedding(model_name=model_name)
+
+#         passage_result = next(iter(model.embed(docs, batch_size=6)))
+#         query_result = next(iter(model.query_embed(docs)))
+#         for result in [passage_result, query_result]:
+#             assert result.indices.tolist() == expected_result["indices"]
+
+#             for i, value in enumerate(result.values):
+#                 assert pytest.approx(value, abs=0.001) == expected_result["values"][i]
+#         if is_ci:
+#             delete_model_cache(model.model._model_dir)
+
+
+# def test_parallel_processing():
+#     is_ci = os.getenv("CI")
+#     model = SparseTextEmbedding(model_name="prithivida/Splade_PP_en_v1")
+#     docs = ["hello world", "flag embedding"] * 30
+#     sparse_embeddings_duo = list(model.embed(docs, batch_size=10, parallel=2))
+#     sparse_embeddings_all = list(model.embed(docs, batch_size=10, parallel=0))
+#     sparse_embeddings = list(model.embed(docs, batch_size=10, parallel=None))
+
+#     assert (
+#         len(sparse_embeddings)
+#         == len(sparse_embeddings_duo)
+#         == len(sparse_embeddings_all)
+#         == len(docs)
+#     )
+
+#     for sparse_embedding, sparse_embedding_duo, sparse_embedding_all in zip(
+#         sparse_embeddings, sparse_embeddings_duo, sparse_embeddings_all
+#     ):
+#         assert (
+#             sparse_embedding.indices.tolist()
+#             == sparse_embedding_duo.indices.tolist()
+#             == sparse_embedding_all.indices.tolist()
+#         )
+#         assert np.allclose(sparse_embedding.values, sparse_embedding_duo.values, atol=1e-3)
+#         assert np.allclose(sparse_embedding.values, sparse_embedding_all.values, atol=1e-3)
+
+#     if is_ci:
+#         delete_model_cache(model.model._model_dir)
 
 
 @pytest.fixture
-def bm25_instance():
+def bm25_instance(request):
     ci = os.getenv("CI", True)
-    model = Bm25("Qdrant/bm25", language="english")
+    disable_stemmer = getattr(request, "param", False)
+    model = Bm25("Qdrant/bm25", language="english", disable_stemmer=disable_stemmer)
```

Comment:
it becomes too complicated, we set both language and disable_stemmer here, so when you read it you might think that language="english" is used somehow, however, it does nothing

Reply:
let's have just one test with disable stemmer, rather than test it in every bm25 test